changelogs/unreleased/gh-7931-httpc-params-escaping.md

```diff
@@ -0,0 +1,6 @@
+## feature/lua/http client
+
+* Performed automated percent-encoding of query params passed into URI using a
+  Lua table. Parameters encoded using `uri.QUERY_PART` when `GET`, `DELETE` or
+  `HEAD` HTTP methods are used and `uri.FORM_URLENCODED` in other cases.
+  (gh-7931)
```

Comment:
Nits on the changelog entry.

1. 'using a Lua table' -> 'using the `params` option'.
2. 'other cases. (gh-xxxx)' -> 'other cases (gh-xxxx).'

Reply:
> > 'using a Lua table' -> 'using the params option'.
>
> What's wrong here? We pass a number of parameters as a Lua table.

That's correct, but a bit vague. I highlighted it, because I stuck here myself for some time. Specifying certain parameter name should make things clear.